src/Config/Level/TypeDeclarationLevel.php

```diff
@@ -86,6 +87,8 @@ final class TypeDeclarationLevel
 
         ReturnTypeFromStrictTernaryRector::class,
         ReturnTypeFromReturnDirectArrayRector::class,
+
+        ResponseReturnTypeControllerActionRector::class,
```

Comment:
Let's add controller types early, so it's easier to review effect of `ReturnTypeFromReturnNewRector` in smaller chunks :)

Make use of  https://github.com/rectorphp/rector-symfony/pull/651

Reply:
@TomasVotruba I found a bug in our project, it got invalid returns on return redirect and array, it got `?array`

```diff
-    public function editAction(Request $request, $id = null)
+    public function editAction(Request $request, $id = null): ?array

Applied rules:
 * ResponseReturnTypeControllerActionRector
```

I will look into it.

Reply:
https://github.com/rectorphp/rector-symfony/pull/660